include/connection.h

```diff
@@ -50,10 +49,27 @@ typedef struct NodeConnectionEntry
 } NodeConnectionEntry;
 
 
+
+typedef struct {
+	ShardId shardId;
+	int nReplicas;
+	bool* status;
```

Comment:
Could you use an enum for the status? (e.g. TRANSACTION_OPEN, TRANSACTION_PREPARED).